.github/workflows/checks.yaml

```diff
@@ -71,33 +71,32 @@ jobs:
       - name: target
         run: "rustc -vV | sed -n 's|host: ||p'"
         if: matrix.os != 'windows-2022'
-      - name: Install cargo-llvm-cov
-        uses: taiki-e/install-action@56ab7930c591507f833cbaed864d201386d518a8
-        with:
-          tool: cargo-llvm-cov
-      - name: Install cargo-nextest
-        uses: taiki-e/install-action@56ab7930c591507f833cbaed864d201386d518a8
+      - name: Install cargo-llvm-cov & cargo-nextest
+        uses: taiki-e/install-action@a48a50298f98c47e46a957ae6f82c44cc4878e42
         with:
-          tool: cargo-nextest
+          tool: cargo-llvm-cov,cargo-nextest
       - name: cargo build
         run: cargo build --target ${{matrix.target}} ${{ matrix.features }}
-      - name: cargo nextest # reports segfaults in a helpful way
-        run: cargo nextest run --target ${{matrix.target}} ${{ matrix.features }} --no-fail-fast
+      - name: cargo llvm-cov nextest
+        if: matrix.os != 'windows-2022'
+        run: cargo llvm-cov nextest --target ${{matrix.target}} ${{ matrix.features }} --lcov --output-path lcov.info
         env:
           RUST_BACKTRACE: 1
-      - name: cargo test with llvm-cov
-        run: cargo llvm-cov --target ${{matrix.target}} ${{ matrix.features }} --lcov --output-path lcov.info
+      - name: cargo nextest
+        if: matrix.os == 'windows-2022'
+        run: cargo nextest run --target ${{matrix.target}} ${{ matrix.features }} --no-fail-fast
         env:
           RUST_BACKTRACE: 1
-        if: matrix.os != 'windows-2022'
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@0565863a31f2c772f9f0395002a31e3f06189574 # v5.4.0
-        if: matrix.os != 'windows-2022'
+        if: matrix.os != 'windows-2022' && matrix.rust == 'stable'
         with:
-          files: lcov.info
+          files: ./lcov.info
           fail_ci_if_error: false
+          flags: test-${{matrix.target}}
           token: ${{ secrets.CODECOV_TOKEN }}
-          slug: codecov/codecov-demo
+          name: test-${{matrix.target}}
+          verbose: true
```

Comment:
- the slug is not needed for us (and I believe it caused issues)
- the flags allow us to see coverage per-target, the unique name makes it easier to see what got uploaded
- may as well make this verbose, easier to troubleshoot future issues